include/unifex/at_coroutine_exit.hpp

```diff
@@ -140,6 +149,15 @@ struct _die_on_done_rec {
       UNIFEX_ASSERT(!"A cleanup action tried to cancel. Calling terminate...");
       std::terminate();
     }
+
+    template(typename CPO)
+      (requires is_receiver_query_cpo_v<CPO> AND
+                is_callable_v<CPO, const Receiver&>)
+    friend auto tag_invoke(CPO cpo, const type& p)
+        noexcept(is_nothrow_callable_v<CPO, const Receiver&>)
+        -> callable_result_t<CPO, const Receiver&> {
+      return cpo(p.rec_);
+    }
```

Comment:
I don't disagree, but I think the CRTP base thing can be overdone. I find more templates and more (source-level) indirection can make the code hard to read and error messages worse. If it's just one function, I will tend to just write the function rather than go the CRTP route.